src/app/message/message.component.ts

```diff
@@ -0,0 +1,219 @@
+import { faMicrophone, faPaperPlane, faEraser, faArrowDown, faFile} from '@fortawesome/free-solid-svg-icons';
+declare var webkitSpeechRecognition: any;
+declare var speechSynthesis: any;
+
+@Component({
+  selector: 'app-message',
+  templateUrl: './message.component.html',
+  styleUrls: ['./message.component.css'],
+})
+export class MessageComponent implements OnInit {
+  allMessages: any[] = [];
+  newMessage: string = '';
+  loginInput: string = '';
+  passwordInput: string = '';
+  voiceMessage: string = '';
+  showScrollButton: boolean = false; 
+  isRecording: boolean = false; 
+  isTyping: boolean = false;
+  iconMicrophone = faMicrophone;
+  iconSend = faPaperPlane;
+  iconClear = faEraser;
+  iconSkipToNew = faArrowDown;
+  iconNew = faFile;
+  isLoginPopupOpen: boolean = false;
+  isLoginSuccessful: boolean = false;
+  isLoginError: boolean = false;
+
+
+  @ViewChild('messageContainer') private messageContainer?: ElementRef;
+
+  constructor(private messageApiService: MessageApiService,
+              private cdr: ChangeDetectorRef
+              ) {}
+
+  ngOnInit() {
+    this.loadMessage();
+    this.isLoginPopupOpen = true;
+  }
+
+  loadMessage() {
+    this.messageApiService.getMessage().subscribe(
+      (response) => {
+        this.allMessages = response.messages;
+        this.readLastAssistantMessage();
+      },
+      (error) => {
+        console.error('Error loading messages:', error);
+      }
+    );
+  }
+
+  sendMessage() {
+    this.messageApiService.addMessage(this.newMessage).subscribe((response) => {
+      if (response.success) {
+        this.loadMessage();
+        this.stopTypingAnimation();
+      } else {
+        console.error('Failed to add message.');
+      }
+    });
+
+    this.addNewMessageToAllMessages();
+
+    setTimeout(() => {
+      this.startTypingAnimation();
+      this.clearTextarea();
+    }, 1000);
+
+    this.clearTextarea();
+  }
+
+  addNewMessageToAllMessages() {
+    const newMessage = {
+      message: this.newMessage,
+      role: "user",
+    };  
+    this.allMessages.push(newMessage);
+  }
+
+  clearTextarea() {
+    this.newMessage = '';
+  }
+
+  startTypingAnimation() {
+    this.isTyping = true;
+    this.cdr.detectChanges();
+  }
+
+  stopTypingAnimation() {
+    this.isTyping = false;
+    this.cdr.detectChanges();
+  }
+
+  @HostListener('window:scroll', ['$event'])
+  onScroll() {
+    this.toggleScrollButtonVisibility();
+  }
+
+  scrollToBottom(): void {
+    if (this.messageContainer) {
+      this.messageContainer.nativeElement.scrollTop = this.messageContainer.nativeElement.scrollHeight;
+    }
+  }
+
+  toggleScrollButtonVisibility(): void {
+    if (this.messageContainer) {
+      this.showScrollButton = this.messageContainer.nativeElement.scrollTop < (this.messageContainer.nativeElement.scrollHeight - this.messageContainer.nativeElement.clientHeight);
+    }
+  }
+
+  isUserMessage(message: any): boolean {
```

Comment:
Należy zdefiniować klasę message, a następnie użyć ja zamiast any - any to jest ostateczność.